### ISSUE TITLE
Spacy is not saving Models with new labels.

### ISSUE THREAD

**Issue by phrul** [explosion/spaCy#6484]
I am using the examples from spacy's documentation to add new entity labels.

I am loading the model like this :

def main(model="en_core_web_md", new_model_name="sak_agreements", output_dir="./sak")

and then using the example code from spacy's site and update model with nlp.update functions. My models gets created and saved to the directory but when I run print(ent.label_, ent.text) it gives me no results.

The problem is the entire code as described on spacy's website works just fine if I replace it with blank 'en' model.

I need to use both "en_core_web_md" models and my own saved models for custom NER lables.

## Your Environment

* Operating System:  Windows.10.10
* Python Version Used: 3.8.5
* spaCy Version Used: 2.3.1
* Environment Information: Models : en_core_web_md

**Comment by svlandeg** [comment]
Can you provide a minimal working code snippet that shows what you've been trying to do, what the results are and what you expected to see instead?

**Comment by phrul** [comment]
I have placed below the code that I used from Spacy's website (https://github.com/explosion/spaCy/blob/master/examples/training/train_new_entity_type.py)

I made only two changes.

1. Added two labels instead of 1 label as shown in the example

label = "User A"
label1 = "User B"

used the above two labels in TRAIN_DATA list.
then added those labels
ner.add_label(label)
ner.add_label(label1)


2. I changed the main method to -
def main(model="en_core_web_md", new_model_name="saks_agreement", output_dir="./saks_agreement", n_iter=30)

It ran and created the ./saks_agreement dir with no errors. But didn't list the two new entities when I did

def print_entities(text = "Is User A a member of our group"):
   nlp = spacy.load('./saks')
  doc = nlp(text)
  for ent in doc.ents:
      print(ent.label_, ent.text)

prints nothing.


______________________________________________________________________________

Code from spacy's website

from __future__ import unicode_literals, print_function

import plac
import random
import warnings
from pathlib import Path
import spacy
from spacy.util import minibatch, compounding


# new entity label
LABEL = "ANIMAL"

# training data
# Note: If you're using an existing model, make sure to mix in examples of
# other entity types that spaCy correctly recognized before. Otherwise, your
# model might learn the new type, but "forget" what it previously knew.
# https://explosion.ai/blog/pseudo-rehearsal-catastrophic-forgetting
TRAIN_DATA = [
    (
        "Horses are too tall and they pretend to care about your feelings",
        {"entities": [(0, 6, LABEL)]},
    ),
    ("Do they bite?", {"entities": []}),
    (
        "horses are too tall and they pretend to care about your feelings",
        {"entities": [(0, 6, LABEL)]},
    ),
    ("horses pretend to care about your feelings", {"entities": [(0, 6, LABEL)]}),
    (
        "they pretend to care about your feelings, those horses",
        {"entities": [(48, 54, LABEL)]},
    ),
    ("horses?", {"entities": [(0, 6, LABEL)]}),
]


@plac.annotations(
    model=("Model name. Defaults to blank 'en' model.", "option", "m", str),
    new_model_name=("New model name for model meta.", "option", "nm", str),
    output_dir=("Optional output directory", "option", "o", Path),
    n_iter=("Number of training iterations", "option", "n", int),
)
def main(model=None, new_model_name="animal", output_dir=None, n_iter=30):
    """Set up the pipeline and entity recognizer, and train the new entity."""
    random.seed(0)
    if model is not None:
        nlp = spacy.load(model)  # load existing spaCy model
        print("Loaded model '%s'" % model)
    else:
        nlp = spacy.blank("en")  # create blank Language class
        print("Created blank 'en' model")
    # Add entity recognizer to model if it's not in the pipeline
    # nlp.create_pipe works for built-ins that are registered with spaCy
    if "ner" not in nlp.pipe_names:
        ner = nlp.create_pipe("ner")
        nlp.add_pipe(ner)
    # otherwise, get it, so we can add labels to it
    else:
        ner = nlp.get_pipe("ner")

    ner.add_label(LABEL)  # add new entity label to entity recognizer
    # Adding extraneous labels shouldn't mess anything up
    ner.add_label("VEGETABLE")
    if model is None:
        optimizer = nlp.begin_training()
    else:
        optimizer = nlp.resume_training()
    move_names = list(ner.move_names)
    # get names of other pipes to disable them during training
    pipe_exceptions = ["ner", "trf_wordpiecer", "trf_tok2vec"]
    other_pipes = [pipe for pipe in nlp.pipe_names if pipe not in pipe_exceptions]
    # only train NER
    with nlp.disable_pipes(*other_pipes), warnings.catch_warnings():
        # show warnings for misaligned entity spans once
        warnings.filterwarnings("once", category=UserWarning, module='spacy')

        sizes = compounding(1.0, 4.0, 1.001)
        # batch up the examples using spaCy's minibatch
        for itn in range(n_iter):
            random.shuffle(TRAIN_DATA)
            batches = minibatch(TRAIN_DATA, size=sizes)
            losses = {}
            for batch in batches:
                texts, annotations = zip(*batch)
                nlp.update(texts, annotations, sgd=optimizer, drop=0.35, losses=losses)
            print("Losses", losses)

    # test the trained model
    test_text = "Do you like horses?"
    doc = nlp(test_text)
    print("Entities in '%s'" % test_text)
    for ent in doc.ents:
        print(ent.label_, ent.text)

    # save model to output directory
    if output_dir is not None:
        output_dir = Path(output_dir)
        if not output_dir.exists():
            output_dir.mkdir()
        nlp.meta["name"] = new_model_name  # rename model
        nlp.to_disk(output_dir)
        print("Saved model to", output_dir)

        # test the saved model
        print("Loading from", output_dir)
        nlp2 = spacy.load(output_dir)
        # Check the classes have loaded back consistently
        assert nlp2.get_pipe("ner").move_names == move_names
        doc2 = nlp2(test_text)
        for ent in doc2.ents:
            print(ent.label_, ent.text)


if __name__ == "__main__":
    plac.call(main)

**Comment by phrul** [comment]
Hi, do u have any update on this?

**Comment by svlandeg** [comment]
What is the result when you print the labels in the NER model?
```
nlp = spacy.load('./saks')
print(nlp.get_pipe("ner").labels)
```
And what is the result if you run your trained model on an example sentence that is in fact expected to contain named entities?
```
test_text = "Do you like horses?"
doc = nlp(test_text)
print("Entities in '%s'" % test_text)
```
You can check the output of this last code snippet after training as well as after loading the model back in from file - it should show an `ANIMAL` entity. However in your code, I don't see any training data for your other label `VEGETABLE`, so the NER model won't be able to recognize vegetable entities. But it should still have it as a label in `nlp.get_pipe("ner").labels`.

**Comment by phrul** [comment]
Thanks. This issues can be closed.
